batch/launch_job.py

```diff
@@ -0,0 +1,139 @@
+    config['nsimulations'] = sims_per_job
+
+    if parallelize_scenarios:
+        scenarios = config['interventions']['scenarios']
+        for s in scenarios:
+            scenario_job_name = f"{job_name}_{s}"
+            config['interventions']['scenarios'] = [s]
+            with open(config_file, "w") as f:
+                yaml.dump(config, f, sort_keys=False)
+            launch_job_inner(scenario_job_name, config_file, num_jobs, sims_per_job, dvc_target, s3_input_bucket, s3_output_bucket, batch_job_definition, batch_job_queue, memory)
+        config['interventions']['scenarios'] = scenarios
+        with open(config_file, "w") as f:
+            yaml.dump(config, f, sort_keys=False)
+    else:
+        with open(config_file, "w") as f:
+            yaml.dump(config, f, sort_keys=False)
+        launch_job_inner(job_name, config_file, num_jobs, sims_per_job, dvc_target, s3_input_bucket, s3_output_bucket, batch_job_definition, batch_job_queue, memory)
+
+    (rc, txt) = subprocess.getstatusoutput(f"git checkout -b run_{job_name}")
+    print(txt)
+    return rc
+
+
+def launch_job_inner(job_name, config_file, num_jobs, sims_per_job, dvc_target, s3_input_bucket, s3_output_bucket, batch_job_definition, batch_job_queue, memory):
+
+    # Prepare to tar up the current directory, excluding any dvc outputs, so it
+    # can be shipped to S3
+    dvc_outputs = get_dvc_outputs()
+    tarfile_name = f"{job_name}.tar.gz"
+    tar = tarfile.open(tarfile_name, "w:gz")
+    for p in os.listdir('.'):
+        if not (p.startswith(".") or p.endswith("tar.gz") or p in dvc_outputs or p == "batch"):
+            tar.add(p, filter=lambda x: None if x.name.startswith('.') else x)
+    tar.close()
+ 
+    # Upload the tar'd contents of this directory and the runner script to S3 
+    runner_script_name = f"{job_name}-runner.sh"
+    local_runner_script = os.path.join(os.path.dirname(os.path.realpath(__file__)), 'runner.sh')
+    s3_client = boto3.client('s3')
+    s3_client.upload_file(local_runner_script, s3_input_bucket, runner_script_name)
+    s3_client.upload_file(tarfile_name, s3_input_bucket, tarfile_name)
+    os.remove(tarfile_name)
+
+    # Prepare and launch the num_jobs via AWS Batch.
+    model_data_path = f"s3://{s3_input_bucket}/{tarfile_name}"
+    results_path = f"s3://{s3_output_bucket}/{job_name}"
+    env_vars = [
+            {"name": "CONFIG_PATH", "value": config_file},
+            {"name": "S3_MODEL_DATA_PATH", "value": model_data_path},
+            {"name": "DVC_TARGET", "value": dvc_target},
+            {"name": "DVC_OUTPUTS", "value": " ".join(dvc_outputs)},
+            {"name": "S3_RESULTS_PATH", "value": results_path},
+            {"name": "SIMS_PER_JOB", "value": str(sims_per_job)}
+    ]
+    s3_cp_run_script = f"aws s3 cp s3://{s3_input_bucket}/{runner_script_name} $PWD/run-covid-pipeline"
+    command = ["sh", "-c", f"{s3_cp_run_script}; /bin/bash $PWD/run-covid-pipeline"]
+    vcpus = min(72, max(2, sims_per_job))
```

Comment:
nit: document magic number 72

Reply:
Just took this out in favor of another CLI param